### DISCUSSION TITLE
Custom ttl function

### DISCUSSION THREAD

**Discussion by emilioestebanez** [caching-tools/next-shared-cache#231] [Ideas]
Right now, if we use the Redis/LRU handlers with `useTtl: true`, then the data is stored for the same amount of time as the revalidation value from Next.js.

I'd like to be able to use stale-while-revalidate with a ttl for stale data (without having to disable `useTtl` because that would store stale data infinitely).

For example if I use `revalidate: 60`, I'd like to use a ttl of 90. That way it'd behave like `Cache-Control: max-age=60 stale-while-revalidate=30`:
- At time = 0, it'll store the data in the cache
- Up to time = 60, it'll directly get the data from the cache
- From time = 60 until time = 90, it'll get the stale data from the cache, and it'll get fresh data in the background
- After time = 90, there is no fresh nor stale data, so it has to fetch it from the origin directly

Is it possible to configure a ttl time with a custom function in the options? For example, something like:

```
(ttl: number) => ttl * 1.5
```

**Comment by better-salmon** [comment]
Hi @emilioestebanez, I apologize for the delay in my response. I believe your idea is feasible, and I plan to test it soon. Thank you!

**Reply by emilioestebanez** [reply]
No worries! Thanks to you for considering it and for your work in this library!

**Comment by better-salmon** [comment]
@emilioestebanez Hi there! I've created PR #249 with an emulation of the SWR strategy. Could you take a look and let me know what you think?

**Reply by emilioestebanez** [reply]
Hi @better-salmon , sorry for the delay. I see this is already part of 0.6.4 I tested it and it works great! Thanks for your quick reaction.
